src/gstreamer/gstkvssink.h

```diff
@@ -147,24 +151,44 @@ struct _KvsSinkCustomData {
             pts_base(0),
             media_type(VIDEO_ONLY),
             first_video_frame(true),
+            onFirstFrame(true),
             frame_count(0),
             first_pts(GST_CLOCK_TIME_NONE),
             producer_start_time(GST_CLOCK_TIME_NONE) {}
     std::unique_ptr<KinesisVideoProducer> kinesis_video_producer;
     std::shared_ptr<KinesisVideoStream> kinesis_video_stream;
 
     std::unordered_set<uint64_t> track_cpd_received;
-    GstKvsSink *kvsSink;
+    GstKvsSink *kvsSink = nullptr;
     MediaType media_type;
     bool first_video_frame;
     uint32_t frame_count;
+    bool onFirstFrame;
 
     std::atomic_uint stream_status;
 
     uint64_t last_dts;
     uint64_t pts_base;
     uint64_t first_pts;
     uint64_t producer_start_time;
+    uint64_t startTime;  // [nanoSeconds]
+    guint errSignalId = 0;
```

Comment:
Are we listening in on stream errors? If not, this should be removed, doesnt seem to be being used anywhere.